autoprotocol/builders.py

```diff
@@ -1738,6 +1747,81 @@ def blowout(self, volume, initial_z, flowrate=None):
             "flowrate": flowrate
         }
 
+    def mode(self, transports=None, mode=None):
```

Comment:
I'm also with Dylan here. I had a hard time trying to figure out what `mode` should default to until I realize that (if I understand correctly), this function should return what mode we should be using. As such, it shouldn't include a `mode` argument

Reply:
mode should default to `air_displacement` for `air` `default` or `none` liquid class types; and `positive_displacement` for `viscous` liquid class type. Currently, we don't specify what mode to use, because `air_displacement` is being hard-coded to be used. However, a user should be able to specify `mode` if needed, and if not specified, we should be able to resolve `mode` to whatever it fits... e.g. if a user wants to transfer glycerol, it should default to using `positive_displacement`, unless specified otherwise. This `mode` method allows for automatic resolution of mode depending on the liquid_class. Also, if say, glycerol is 'viscous' type, but if diluted enough, can be transferred using `air_displacement`. Hence, this will kick in only `if mode is None`. If user has a `mode` specified, that should be respected. Also, if (by any chance) there are mutiple `mode` specified in one liquid handle, that should be flagged... The behavior could also depend on how we add such aliquot propeties(?), tip types (since air-displacement and positive-displacement will use different tip types), etc, in the future as well....

Reply:
So if they specify a mode here, we want to check that this mode does not clash with their liquid types?

Reply:
Maybe `desired_mode` would be more descriptive.

Reply:
Right. I'll rename the method, and will add more documentation as Josh requested.